### PR TITLE
Feat(core): add ability to clear stencil buffer

### PR DESCRIPTION
**Pull request by cptbtptpbcptdtptp**
export enum CameraClearFlags {
  /* Do nothing. */
  None = 0,
  /* Clear color with scene background. */
  Color = 1,
  /* Clear depth only. */
  Depth = 2,
  /* Clear depth only. */
  Stencil = 4,

  /* Clear color with scene background and depth. */
  ColorDepth = 3,
  /* Clear color with scene background and stencil. */
  ColorStencil = 5,
  /* Clear depth and stencil. */
  DepthStencil = 6,

  /* Clear color with scene background, depth, and stencil. */
  All = 7
}

### CODE REVIEW

File: packages/core/src/input/pointer/PointerManager.ts [hunk 1]
```diff
@@ -244,7 +244,7 @@ export class PointerManager {
           // TODO: Only check which colliders have listened to the input.
           if (this._engine.physicsManager.raycast(camera.viewportPointToRay(point, ray), hitResult)) {
             return hitResult.entity;
-          } else if (camera.clearFlags === CameraClearFlags.DepthColor) {
+          } else if (camera.clearFlags === CameraClearFlags.Color) {
```
Comment (GuoLei1990) [inline packages/core/src/input/pointer/PointerManager.ts:247]: Maybe should `camera.clearFlags & CameraClearFlags.Color`?

File: packages/core/src/enums/CameraClearFlags.ts [hunk 2]
```diff
@@ -2,10 +2,22 @@
...
+  /* Clear color and depth from background. */
+  ColorDepth = 3,
+  /* Clear color and stencil from background. */
+  ColorStencil = 5,
+  /* Clear depth and stencil from background. */
+  DepthStencil = 6,
+
+  /** Clear stencil, depth and color from background. */
+  All = 7
 }
```
[the remark this answers is not in the document]
Comment (cptbtptpbcptdtptp) [reply packages/core/src/enums/CameraClearFlags.ts:23]: Done.

File: packages/core/src/input/pointer/PointerManager.ts (hunk 1 above)
Comment (cptbtptpbcptdtptp) [reply packages/core/src/input/pointer/PointerManager.ts]: Done.

**Review by GuoLei1990 — APPROVED** [review]
+1

**Pull request merged**